core/domain/feature_gating_services.py

```diff
@@ -0,0 +1,191 @@
+            be evaluated.
+
+    Returns:
+        bool. The value of the feature flag, True if it's enabled.
+    """
+    context = _create_evaluation_context_for_server()
+    values_dict = _evaluate_feature_flag_values_for_context(
+        [feature_name], context)
+    return values_dict[feature_name]
+
+
+def update_feature_flag_rules(
+        feature_name, committer_id, commit_message, new_rule_dicts):
+    """Updates the feature flag's rules.
+
+    Args:
+        feature_name: str. The name of the feature to update.
+        committer_id: str. ID of the committer.
+        commit_message: str. The commit message.
+        new_rule_dicts: list(dist). A list of dict mappings of all fields
+            of PlatformParameterRule object.
+    """
+    if feature_name not in ALL_FEATURES_NAMES_SET:
+        raise Exception('Feature flag not exist: %s.' % feature_name)
+
+    registry.Registry.update_platform_parameter(
+        feature_name, committer_id, commit_message, new_rule_dicts)
+
+
+# TODO(#10211): Currently Oppia runs in either of the two modes:
+# dev or prod. There should be another mode 'test' added for QA testing,
+# once it is added, this function needs to be updated to take that into
+# consideration.
+def _get_server_mode():
+    """Returns the running mode of Oppia.
+
+    Returns:
+        str. The server mode of Oppia, 'dev' if Oppia is running in development
+        mode, 'prod' if in production mode.
+    """
+    if constants.DEV_MODE:
+        return param_domain.SERVER_MODES.dev
+    else:
+        return param_domain.SERVER_MODES.prod
+
+
+def _create_evaluation_context_for_server():
+    """Returns evaluation context with information of the server.
+
+    Returns:
+        EvaluationContext. The context for evaluation.
+    """
+    return param_domain.EvaluationContext.from_dict(
+        {
+            'client_type': None,
+            'browser_type': None,
+            'app_version': None,
+            'user_locale': None,
+        },
+        {
+            'server_mode': _get_server_mode()
+        }
+    )
+
+
+def _evaluate_feature_flag_values_for_context(feature_names, context):
+    """Evaluates and returns the values for specified feature flags.
+
+    Args:
+        feature_names: list(str). The names of feature flags that need to
+            be evaluated.
+        context: EvaluationContext. The context used for evaluation.
+
+    Returns:
+        dict. The keys are the feature names and the values are boolean
+        results of corresponding flags.
+    """
+    unknown_feature_names = []
+    for feature_name in feature_names:
+        if feature_name not in ALL_FEATURES_NAMES_SET:
+            unknown_feature_names.append(feature_name)
+    if len(unknown_feature_names) > 0:
+        raise Exception(
+            'Unknown feature flag(s): %s.' % unknown_feature_names)
+
+    result_dict = {}
+    for feature_name in feature_names:
+        flag_domain = registry.Registry.get_platform_parameter(
```

Comment:
Will this make `n` datastore calls? Can we use `get_multi` and fetch all at once? (not sure!)

Reply:
But still, it will be `n` datastore call, right? is there any limitation on `len(feature_names)`? Can we do it better? 

[I'm fine with having it this way, just wanted to check whether we can do it better?]

Reply:
By 'datastore call' do you mean accessing the storage? Since I am using memcache in `get_platform_parameter`, in most cases there won't be any calls to the datastore.